network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -701,7 +702,34 @@ def get_status_for_airtable(self):
 
     @property
     def original_product(self):
-        return get_original_by_slug(ProductPage, self.slug)
+        try:
+            original = get_original_by_slug(ProductPage, self.slug)
+
+        except ProductPage.DoesNotExist as error:
+            """
+            This may happen when a product was created, got localized,
+            then the original product was deleted a new product with the
+            same title (and thus, slug) gets created.
+
+            The old localizations are still around, so `sync_locale_trees`
+            will see a new page to set up aliases for, but their slug ends
+            up conflicting with the slugs for the original localized pages,
+            and so they get `-1` appended, meaning we can't find their
+            "original" equivalent using their slug: we need to rewrite
+            the slug to remove the number suffix first.
+
+            See: https://github.com/wagtail/wagtail/issues/7592
+
+            TODO: Remove this patch code once #7592 is addressed.
+            """
+            slug = re.sub('(-\d+)+$', '', self.slug)
```

Comment:
Just because this might happen over and over, replace `-1`, `-1-1`, `-1-1-1`, etc. etc.